### ISSUE TITLE
Finish up framebuffer objects

### ISSUE THREAD

**Issue by mikolalysenko** [mikolalysenko/regl#46]
There are lots of gaps in the framebuffer object API which we need to fill in.  Also the constructor is currently untested and allocates a bunch of garbage.  The following needs to happen before we release 1.0:

* We need complete test coverage for framebuffer objects and all code generation pathways associated to them
* There should be a method for quickly resizing or updating a framebuffer.  Calling it repeatedly should have no effect.
* We need maybe 1-2 more good examples showing different uses of framebuffers

**Comment by Erkaman** [comment]
Once framebuffers are finished, I can implement some examples showing usages of framebuffers. I was thinking about implementing a simple blur post-process effect, shadow mapping, and maybe also a simple deferred renderer.

**Comment by Erkaman** [comment]
This is not finished yet though. We also need some examples.

**Comment by Erkaman** [comment]
I will work on that when I have the time.
